### ISSUE TITLE
Amplify/boost mic volume

### ISSUE THREAD

**Issue by Brottweiler** [MaartenBaert/ssr#392]
Requesting a feature where you, in my case with JACK, would be able to amplify or boost the microphone volume. This seems to be an option in OBS, and would be useful for me that has a USB microphone (Blue Snowball) and is using it with JACK but has pretty low volume.

Haven't found a way to boost it with ALSA/JACK. :disappointed:

**Comment by rauldipeas** [comment]
I do this volume boost on post edit of video, with gate, compression and normalize audio effects, but it will be very welcome to be an option in SSR on the future releases.

+1

**Comment by Brottweiler** [comment]
@rauldipeas yeah, recording with audacity and amplify would be a good way, but I'm mostly thinking of livestreaming.

**Comment by rauldipeas** [comment]
On live streaming You can use a plugin rack to process audio in realtime, but, You will need lowlatency kernel, Jack audio server, Carla plugins rack, some ladspa/lv2 plugins and alot of patience to figure out how to do all of this things work together.

It's hard, but possible.

**Comment by Brottweiler** [comment]
Well I am using the linux-ck kernel so I think it is lowlatency, and I use JACK, but yeah, probably not worth the trouble for me.

**Comment by trebmuh** [comment]
IIRC you can have a boost plugin with .asoundrc (ALSA config file)

**Comment by Brottweiler** [comment]
@trebmuh It would be helpful to know how to do this.

**Comment by trebmuh** [comment]
@Brottweiler : if I remember correctly, it came from : http://alsa.opensrc.org/
Reading at the 3 pages about "ladspa" there, it's about including a ladspa plugin in the asoundrc configuration. Never did that myself, mentioning it here if you're feeling like a adventurer.

**Comment by rauldipeas** [comment]
There's "veromix" too, it's simplify this process.

**Comment by iamgreaser** [comment]
jack-rack + an amplifier LADSPA plugin should be fine. I run FreeBSD and have no issues with latency on JACK, and it doesn't even use realtime mode (I use a buffer size of 128 w/ 3 buffers @ 48kHz which is apparently 8ms of latency).

Suggested setup for JACK: launch from qjackctl, use 3 buffers, set a suitable buffer size (512 is a good starting point).

**Comment by MaartenBaert** [comment]
Before you try anything else, run 'alsamixer' in a terminal and check whether your sound card has some microphone gain/boost options that you haven't set to maximum yet. Most sound cards have a 'gain' setting (which is the usual volume slider) and also a 'mic boost' setting which is less accurate but useful if the normal gain isn't enough. USB microphones might be different though. In that case amplifying through JACK is a good idea (jack-rack + LADSPA plugin, as suggested).

**Comment by Brottweiler** [comment]
@MaartenBaert ![](https://i.imgur.com/kxQvIrE.png)

Yeah, this is the problem with USB mics under linux.

I am using JACK and I would like to try the LADSPA thing, it's just confusing to me right now, not sure how to set it up.

**Comment by Brottweiler** [comment]
I've looked into jack-rack and don't quite understand it, and I can't really find any good documentation on it.

I can add mono or sterio amplification in jack-rack, and that's straight forward. I can up the amplification from 1, and enable it. Then I *assume* I have to go into **connections** in JACK and connect the jack-rack to... whatever is capturing, right? 

So if I started to record (or it seems like i can just open SSR), I should be able to go into Connections, then connect the jack-rack output to SSR, then up the amplification slider in jack-rack, and that'll do it? I doesn't seem to work that way though... Unless I am doing something wrong. Any ideas?

**Comment by rauldipeas** [comment]
Maybe, if You try Carla Rack from KXStudio repositories, will be more easy to work, Carla supports LADSPA, LV2, VST(native and Wine), DSSI and embedded Disthro plugins, and comes with Catia patchbay for make connections between audio signal flows.

**Comment by rauldipeas** [comment]
I think that it's what the SSR needs for this purpose, a integration with this kind of tools, a simple button to call them on the interface of the SSR will be very helpful.

**Comment by iamgreaser** [comment]
I think what SSR needs is the ability to record video well. Which is what it does already.

It seems you've understood jack-rack properly.

All you need to do now is pipe system input into jack_rack input, jack_rack output into SSR output, and then disconnect system input from SSR output.

So you've pretty much got it.

**Comment by rauldipeas** [comment]
@iamgreaser This issue is about "audio" possibilities on SSR, I think this video capability is very well at this time.

**Comment by Brottweiler** [comment]
@iamgreaser 
>system input into jack_rack input, jack_rack output into SSR output, and then disconnect system input from SSR output

I am a little confused about this. The left panel is output, and the right one is input. You cannot pile input into input. Did you write it correctly? Here's a screenshot

![](https://i.imgur.com/07iwEZ4.png)

**Comment by Brottweiler** [comment]
I think what you meant was this

1. system output into jack_rack input
2. jack_rack output into SSR input
3. disconnect system input from SSR output

I recorded and upped the amplification while saying "test", and it worked!

**Comment by rauldipeas** [comment]
@Brottweiler Catia Patchbay it's much more easy than this.
